emulator.py

```diff
@@ -421,6 +422,155 @@ def object_get(bucket_name, object_name):
     return blob.rest_media(flask.request)
 
 
+# === OBJECT SPECIAL OPERATIONS === #
+
+
+@gcs.route("/b/<bucket_name>/o/<path:object_name>/compose", methods=["POST"])
+@retry_test(method="storage.objects.compose")
+def objects_compose(bucket_name, object_name):
+    bucket = db.get_bucket_without_generation(bucket_name, None).metadata
+    payload = json.loads(flask.request.data)
+    source_objects = payload.get("sourceObjects", None)
+    if source_objects is None:
+        testbench.error.missing("source component", None)
+    if len(source_objects) > 32:
+        testbench.error.invalid(
+            "The number of source components provided (%d > 32)" % len(source_objects),
+            None,
+        )
+    composed_media = b""
+    for source_object in source_objects:
+        print("%s\n" % source_object)
```

Comment:
The code refactor here looks much more clear. Thanks!
Is this print statement for testing purposes?

Reply:
Doh! I was supposed to remove that. Done.